File: backend/app/models/projekt.py
from enum import Enum
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class ProjektFaza(str, Enum):
    RFQ = "RFQ"
    PONUDBA = "Ponudba"
    NAROCILO = "Naročilo"
    TEHNOLOGIJA = "Tehnologija"
    NABAVA = "Nabava"
    PROIZVODNJA = "Proizvodnja"
    DOSTAVA = "Dostava"
    ZAKLJUCEK = "Zaključek"


class ProjektStatus(str, Enum):
    AKTIVEN = "Aktiven"
    NA_CAKANJU = "Na čakanju"
    ZAKLJUCEN = "Zaključen"
    PREKLICAN = "Preklican"


class ProjektBase(BaseModel):
    naziv: str
    stranka_id: Optional[int] = None
    opombe: Optional[str] = None


class ProjektCreate(ProjektBase):
    pass


class ProjektUpdate(BaseModel):
    naziv: Optional[str] = None
    faza: Optional[ProjektFaza] = None
    status: Optional[ProjektStatus] = None
    opombe: Optional[str] = None
    odgovorni_prodaja: Optional[int] = None
    odgovorni_tehnolog: Optional[int] = None


class Projekt(ProjektBase):
    id: int
    stevilka_projekta: str
    faza: ProjektFaza = ProjektFaza.RFQ
    status: ProjektStatus = ProjektStatus.AKTIVEN
    datum_rfq: datetime
    datum_zakljucka: Optional[datetime] = None
    odgovorni_prodaja: Optional[int] = None
    odgovorni_tehnolog: Optional[int] = None

    class Config:
        from_attributes = True


class ProjektCasovnicaEvent(str, Enum):
    USTVARJEN = "Ustvarjen"
    SPREMEMBA_FAZE = "Sprememba faze"
    SPREMEMBA_STATUSA = "Sprememba statusa"
    NOV_DOKUMENT = "Nov dokument"
    NOV_EMAIL = "Nov email"
    DN_USTVARJEN = "DN ustvarjen"
    DN_STATUS = "DN status"
    CQ_RFQ = "CQ RFQ vnesen"
    CQ_SYNC = "CQ sinhronizacija"
    KOMENTAR = "Komentar"


class ProjektCasovnica(BaseModel):
    id: int
    projekt_id: int
    dogodek: ProjektCasovnicaEvent
    opis: str
    stara_vrednost: Optional[str] = None
    nova_vrednost: Optional[str] = None
    datum: datetime
    uporabnik_ali_agent: str

    class Config:
        from_attributes = True

File: flutter_app/lib/models/projekt.dart
/// Model projekta
class Projekt {
  final int id;
  final String stevilkaProjekta;
  final String naziv;
  final int? strankaId;
  final String faza;
  final String status;
  final DateTime datumRfq;
  final DateTime? datumZakljucka;
  final String? opombe;

  Projekt({
    required this.id,
    required this.stevilkaProjekta,
    required this.naziv,
    this.strankaId,
    required this.faza,
    required this.status,
    required this.datumRfq,
    this.datumZakljucka,
    this.opombe,
  });

  factory Projekt.fromJson(Map<String, dynamic> json) {
    return Projekt(
      id: json['id'],
      stevilkaProjekta: json['stevilka_projekta'],
      naziv: json['naziv'],
      strankaId: json['stranka_id'],
      faza: json['faza'],
      status: json['status'],
      datumRfq: DateTime.parse(json['datum_rfq']),
      datumZakljucka: json['datum_zakljucka'] != null
          ? DateTime.parse(json['datum_zakljucka'])
          : null,
      opombe: json['opombe'],
    );
  }

  /// Vrne barvo glede na fazo
  String get fazaColor {
    switch (faza) {
      case 'RFQ':
        return '#3B82F6'; // blue
      case 'Ponudba':
        return '#8B5CF6'; // purple
      case 'Naročilo':
        return '#10B981'; // green
      case 'Tehnologija':
        return '#F59E0B'; // amber
      case 'Nabava':
        return '#EF4444'; // red
      case 'Proizvodnja':
        return '#06B6D4'; // cyan
      case 'Dostava':
        return '#84CC16'; // lime
      case 'Zaključek':
        return '#6B7280'; // gray
      default:
        return '#6B7280';
    }
  }
}

/// Model časovnice
class ProjektCasovnica {
  final int id;
  final int projektId;
  final String dogodek;
  final String opis;
  final String? staraVrednost;
  final String? novaVrednost;
  final DateTime datum;
  final String uporabnikAliAgent;

  ProjektCasovnica({
    required this.id,
    required this.projektId,
    required this.dogodek,
    required this.opis,
    this.staraVrednost,
    this.novaVrednost,
    required this.datum,
    required this.uporabnikAliAgent,
  });

  factory ProjektCasovnica.fromJson(Map<String, dynamic> json) {
    return ProjektCasovnica(
      id: json['id'],
      projektId: json['projekt_id'],
      dogodek: json['dogodek'],
      opis: json['opis'],
      staraVrednost: json['stara_vrednost'],
      novaVrednost: json['nova_vrednost'],
      datum: DateTime.parse(json['datum']),
      uporabnikAliAgent: json['uporabnik_ali_agent'],
    );
  }
}
